src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs

```diff
@@ -312,58 +315,89 @@ pub fn caml_pasta_fp_plonk_proof_example_with_foreign_field_mul(
-    external_checks
-        .extend_witness_bound_addition(&mut witness, &foreign_field_modulus.to_field_limbs());
+    // Result compact-multi-range-check
+    CircuitGate::extend_compact_multi_range_check(&mut gates, &mut next_row);
+    gates.connect_cell_pair((1, 0), (4, 1)); // remainder01
+    gates.connect_cell_pair((1, 1), (2, 0)); // remainder2
+    external_checks.extend_witness_compact_multi_range_checks(&mut witness);
+    // These are the coordinates (row, col) of the remainder limbs in the witness
+    // remainder0 -> (3, 0), remainder1 -> (4, 0), remainder2 -> (2,0)
 
-    // Left input multi-range-check
+    // Constant single Generic gate for result bound
+    CircuitGate::extend_high_bounds(&mut gates, &mut next_row, &foreign_field_modulus);
+    gates.connect_cell_pair((6, 0), (1, 1)); // remainder2
+    external_checks.extend_witness_high_bounds_computation(&mut witness, &foreign_field_modulus);
+
+    // Quotient multi-range-check
     CircuitGate::extend_multi_range_check(&mut gates, &mut next_row);
-    gates.connect_cell_pair((0, 0), (4, 0));
-    gates.connect_cell_pair((0, 1), (5, 0));
-    gates.connect_cell_pair((0, 2), (6, 0));
-    range_check::witness::extend_multi_limbs(&mut witness, &left_input.to_field_limbs());
+    gates.connect_cell_pair((1, 2), (7, 0)); // quotient0
+    gates.connect_cell_pair((1, 3), (8, 0)); // quotient1
+    gates.connect_cell_pair((1, 4), (9, 0)); // quotient2
+                                             // Witness updated below
 
-    // Right input multi-range-check
+    // Multiplication witness value quotient_bound, product1_lo, product1_hi_0 multi-range-check
     CircuitGate::extend_multi_range_check(&mut gates, &mut next_row);
-    gates.connect_cell_pair((0, 3), (8, 0));
-    gates.connect_cell_pair((0, 4), (9, 0));
-    gates.connect_cell_pair((0, 5), (10, 0));
-    range_check::witness::extend_multi_limbs(&mut witness, &right_input.to_field_limbs());
+    gates.connect_cell_pair((1, 5), (11, 0)); // quotient_bound
+    gates.connect_cell_pair((0, 6), (12, 0)); // product1_lo
+    gates.connect_cell_pair((1, 6), (13, 0)); // product1_hi_0
+                                              // Witness updated below
+
+    // Add witness for external multi-range checks:
+    // [quotient0, quotient1, quotient2]
+    // [quotient_bound, product1_lo, product1_hi_0]
+    external_checks.extend_witness_multi_range_checks(&mut witness);
+
+    // DESIGNER CHOICE: left and right (and result bound from before)
+    let left_limbs = left_input.to_field_limbs();
+    let right_limbs = right_input.to_field_limbs();
+    // Constant Double Generic gate for result and quotient bounds
+    external_checks.add_high_bound_computation(&left_limbs[2]);
+    external_checks.add_high_bound_computation(&right_limbs[2]);
+    CircuitGate::extend_high_bounds(&mut gates, &mut next_row, &foreign_field_modulus);
+    gates.connect_cell_pair((15, 0), (0, 2)); // left2
```

Comment:
This code is confusing, which lines are doing double generic for result and quotient VS which are doing left2 and right2?